#########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################

# Basic parameters
mass        : 1.2 # kg 
gra         : 9.81 
ctrl_freq_max   : 100.0
use_bodyrate_ctrl: false
max_manual_vel: 1.0
max_angle: 30  # Attitude angle limit in degree. A negative value means no limit.
low_voltage: 13.2 # 4S battery

# Path following parameters
path_following:
  lookahead_distance: 1.0 # meters
  speed: 1.0 # m/s

# RC reverse settings
rc_reverse:
    roll: false
    pitch: false
    yaw: false
    throttle: false

# Auto takeoff and land
auto_takeoff_land:
    enable: true
    enable_auto_arm: true
    no_RC: false
    takeoff_height: 1.0 # m
    takeoff_land_speed: 0.3 # m/s

# Thrust model
thrust_model:
    print_value: false
    accurate_thrust_model: false
    K1: 0.7583
    K2: 1.6942
    K3: 0.6786
    hover_percentage: 0.30

# SE3 Controller gains
se3_gains:
    # Horizontal gains
    horizontal:
        kp: 3.0
        kv: 2.0
        ka: 1.0
        kiw: 0.1      # world frame integral
        kib: 0.1      # body frame integral
        kiw_lim: 10.0 # integral limit
        kib_lim: 10.0 # integral limit
    
    # Vertical gains
    vertical:
        kp: 15.0
        kv: 8.0
        ka: 1.0
    
    # Attitude gains
    attitude:
        kq_roll_pitch: 3.0
        kq_yaw: 2.0
    
    # Attitude rate gains
    attitude_rate:
        kw_roll_pitch: 4.0
        kw_yaw: 4.0
    
    # Mass estimator
    mass_estimator:
        km: 1.0
        km_lim: 3.0 # kg, at least 1/2 of the UAV mass

# Constraints
constraints:
    throttle_saturation: 0.9 # [-], range from 0 to 1
    tilt_angle_failsafe:
        enabled: true
        limit: 45.0 # [deg]

# Gain filtering
gain_filtering:
    rate: 10.0 # [Hz]
    perc_change_rate: 1.0 # [100%/s]
    min_change_rate: 0.2 # perc of the difference
    gain_mute_coefficient: 0.5 # [-]

# Rampup
rampup:
    enabled: true
    speed: 0.75 # [1/s]

# Message timeouts
msg_timeout:
    odom: 0.5
    rc:   0.5
    path: 0.5
    imu:  0.5
    bat:  0.5